src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -51,6 +56,43 @@ public void saveChallenge(ChallengeDTO challengeDTO) {
         
     }
 
+    @Override
+    public ChallengeDTO updateChallenge(Long challengeId, ChallengeDTO challengeDTO) throws NotFoundException {
+
+        Long userId;
+
+        if(challengeDTO.getChallengeCategory() != null) {
+
+            if (challengeDTO.getChallengeCategory().getId() == null) {
+                challengeDTO.setChallengeCategory(null);
+            } else
+            {
+            challengeCategoryRepository.findById(challengeDTO.getChallengeCategory().getId())
+                    .orElseThrow(() -> new InvalidFieldException().setMessage("Categoria aleasa trebuie sa existe"));
+            }
+        }
+
+        if(!challengeId.equals(challengeDTO.getId()))
+        {
+            throw new InvalidFieldException().setMessage("Id-ul challenge-ului din path trebuie ca corespunda cu cel din DTO");
+        }
+
+        Challenge oldChallenge = challengeRepository.findById(challengeId).orElseThrow(
+                ()->new NotFoundException().setMessage("Challenge-ul " + challengeId +" nu exista")
+        );
+
+        userId = challengeDTO.getCreator().getId();
+
+        if(!oldChallenge.getCreator().getId().equals(userId))
+        {
+            throw new InvalidFieldException().setMessage("Creatorul unui challenge nu se poate schimba");
+        }
+
+       challengeRepository.save(ChallengeMapper.convertChallengeDTOToChallenge(challengeDTO));
+
+        return challengeDTO;
+    }
```

Comment:
Aceasta metoda se numeste updateChalllenge. De ce face mai multe lucruri? Adica verifica ca exista categoria, dupa care seteaza categoria pe challenge, dupa care verifici identitatea creatorului,  dupa care face update la challenge. 
Ar trebui sa faci o metoda ca sa verifice daca exista un challengeCategory pe baza unui id de categori. 
O alta metoda care sa verifice existenta unui user pe baza unui userId. In final definesti o alta metoda, challengeValidation care sa fie responsabila cu validarea unui challengeDto. Aceasta din urma, ar trebui sa le apeleze metoda de verificare a challengeCategory, metoda de validare a userId + alte verificari ce tin de challenge. In final, in metoda updateChallenge se apeleaza validateChallenge, in functie de rezultat se trecere la update-ul efectiv. Adica cauti in baza de date Challenge-ul pe baza id-ul si FACI update pe campurile celui gasit in baza de date. Nu mai face un alt obiect prin apelarea metodei convertChallengeDTOToChallenge ( o alta posibilitate aici, este ca in ChallengeMapper sa faci o metoda de enrichChallenge care sa primeasca un Challenge si un ChallengeDTO si sa faca set pe campurile din Challenge cu valorile corespunzatoare din ChallengeDTO)